src/hivpy/circumcision.py

```diff
@@ -158,6 +162,17 @@ def update_vmmc(self, pop):
                                                 & pop.data[col.CIRCUMCISION_DATE].isnull()]
                 pop.data.loc[new_circ_males, col.CIRCUMCISION_DATE] = self.date
 
+                # those that just got circumcised and weren't tested last time step get tested now
+                just_tested = pop.get_sub_pop([(col.CIRCUMCISION_DATE, op.eq, pop.date),
+                                               (col.LAST_TEST_DATE, op.ne, pop.date - time_step)])
+                # correctly set up related columns
+                if len(just_tested) > 0:
+                    pop.set_present_variable(col.EVER_TESTED, True, just_tested)
+                    pop.set_present_variable(col.LAST_TEST_DATE, pop.date, just_tested)
+                    # "reset" dummy partner columns
+                    pop.set_present_variable(col.NSTP_LAST_TEST, 0, just_tested)
+                    pop.set_present_variable(col.NP_LAST_TEST, 0, just_tested)
```

Comment:
This turned out to be a smaller bit of code than I initially expected, but it should probably still be moved to the HIV testing module.